mpsort-mpi.c

```diff
@@ -326,7 +326,7 @@ checksum(void * base, size_t nbytes, MPI_Comm comm)
 {
     uint64_t sum = 0;
     char * ptr = base;
-    ptrdiff_t i = 0;
```

Comment:
With -Wextra it gives a warning because size_t nbytes might overflow ptrdiff_t

Reply:
OK. But there must be a reason the flag it is called extra..

I've also seen people cast size_t to signed in a loop to avoid the -Wextra warning.
Later i is used as an array indices, which shall be signed.

Reply:
Yes - these warnings make no difference in general. I am just being picky about them because I have seen icc do weird things where signed int overflow is theoretically possible. The other way to do it is to make nbytes ptrdiff_t, which is sensible because that is what it is. I did that instead.